haskell-language-server.cabal

```diff
@@ -108,6 +108,63 @@ library
 
   default-language: Haskell2010
 
+flag ghc-lib
+  default: False
+  manual: True
+  description: Force dependency on ghc-lib-parser even if GHC API in the ghc package is supported
+
+library hlint-plugin
```

Comment:
As noted above the private lib does not work for stack based cradle.
Moreover i have to peridically delete the haskell-language-server library from the cabal store due to 
```
exe\Main.hs:147:9: error:
    • Couldn't match expected type ‘Ide.Types.PluginDescriptor’
                  with actual type ‘haskell-language-server-0.2.0.0:Ide.Types.PluginDescriptor’
      NB: ‘haskell-language-server-0.2.0.0:Ide.Types.PluginDescriptor’
            is defined in ‘Ide.Types’
                in package ‘haskell-language-server-0.2.0.0’
          ‘Ide.Types.PluginDescriptor’
            is defined in ‘Ide.Types’
                in package ‘haskell-language-server-0.2.0.0’
    • In the expression: Hlint.descriptor "hlint"
      In the expression:
        [GhcIde.descriptor "ghcide", Pragmas.descriptor "pragmas",
         Floskell.descriptor "floskell", Ormolu.descriptor "ormolu", ....]
      In an equation for ‘basePlugins’:
          basePlugins
            = [GhcIde.descriptor "ghcide", Pragmas.descriptor "pragmas",
               Floskell.descriptor "floskell", ....]
    |
147 |       , Hlint.descriptor "hlint"
    |         ^^^^^^^^^^^^^^^^^^^^^^^^
```
So i think this plugin must be separated in its own package, after the restructuration of the lib (see #164)